File: erp_mmust/erp_mmust/report/fee_collection_per_vote/fee_collection_per_vote.js
frappe.query_reports["Fee Collection Per Vote"] = {
    "filters": [
        {
            "fieldname": "company",
            "label": __("Company"),
            "fieldtype": "Link",
            "options": "Company",
            "default": frappe.defaults.get_user_default("Company"),
            "reqd": 1
        },
        {
            "fieldname": "from_date",
            "label": __("From Date"),
            "fieldtype": "Date",
            "default": frappe.datetime.add_months(frappe.datetime.get_today(), -1),
            "reqd": 1
        },
        {
            "fieldname": "to_date",
            "label": __("To Date"),
            "fieldtype": "Date",
            "default": frappe.datetime.get_today(),
            "reqd": 1
        },
        {
            "fieldname": "vote",
            "label": __("Vote (Income Account)"),
            "fieldtype": "Link",
            "options": "Account",
            "reqd": 0,
            "get_query": function () {
                return {
                    "filters": {
                        "company": frappe.query_report.get_filter_value('company')
                    }
                };
            }
        }
    ]
};

File: erp_mmust/erp_mmust/report/fee_collection_per_vote/fee_collection_per_vote.py
import frappe
from frappe import _

def execute(filters=None):
    if not filters:
        filters = {}

    columns = get_columns()
    data = get_data(filters)
    
    return columns, data, None, None, None

def get_columns():
    return [
        {
            "label": _("Posting Date"),
            "fieldname": "posting_date",
            "fieldtype": "Date",
            "width": 100
        },
        {
            "label": _("Voucher Type"),
            "fieldname": "voucher_type",
            "fieldtype": "Data",
            "width": 120
        },
        {
            "label": _("Voucher No"),
            "fieldname": "voucher_no",
            "fieldtype": "Dynamic Link",
            "options": "voucher_type",
            "width": 150
        },
        {
            "label": _("Account"),
            "fieldname": "account",
            "fieldtype": "Link",
            "options": "Account",
            "width": 200
        },
        {
            "label": _("Debit"),
            "fieldname": "debit",
            "fieldtype": "Currency",
            "width": 120
        },
        {
            "label": _("Credit"),
            "fieldname": "credit",
            "fieldtype": "Currency",
            "width": 120
        },
        {
            "label": _("Party Type"),
            "fieldname": "party_type",
            "fieldtype": "Data",
            "width": 100
        },
        {
            "label": _("Party"),
            "fieldname": "party",
            "fieldtype": "Dynamic Link",
            "options": "party_type",
            "width": 150
        },
        {
            "label": _("Remarks"),
            "fieldname": "remarks",
            "fieldtype": "Data",
            "width": 200
        }
    ]

def get_data(filters):
    gl_entries = frappe.get_all(
        "GL Entry",
        filters={
            "company": filters.get("company"),
            "account": filters.get("account"),
            "posting_date": ["between", [filters.get("from_date"), filters.get("to_date")]],
        },
        fields=[
            "posting_date",
            "voucher_type",
            "voucher_no",
            "account",
            "debit",
            "credit",
            "party_type",
            "party",
            "remarks"
        ],
        order_by="posting_date"
    )

    return gl_entries
